aiomisc/service/periodic.py

```diff
@@ -29,10 +30,7 @@ async def start(self) -> None:
         log.info("Periodic service %s started", self)
 
     async def stop(self, err: Exception = None) -> None:
-        if self.periodic.task:
-            await self.periodic.task
-
-        await self.periodic.stop()
+        await asyncio.gather(self.periodic.stop(), return_exceptions=True)
```

Comment:
I think plain await of `self.periodic.stop()` wrapped with `suppress` will be cleaner.

Reply:
It won't do the same thing, the suppress will catch the `asyncio.CancelledError` but instead of raising an exception just returns.